core/procedural-primitives/procedural-primitives.zig

```diff
@@ -183,3 +183,94 @@ pub fn createCubePrimitive(allocator: std.mem.Allocator, size: f32) !Primitive {
 
     return Primitive{ .vertex_data = vertex_data, .vertex_count = vertex_count, .index_data = index_data, .index_count = index_count, .type = .quad };
 }
+
+pub fn createCylinderPrimitive(allocator: std.mem.Allocator, radius: f32, height: f32, num_sides: u32) !Primitive {
+
+    var vertex_data = try std.ArrayList(VertexData).initCapacity(allocator, 500);
+    var index_data = try std.ArrayList(u32).initCapacity(allocator, 500);
+    var indexes: u32 = 0;
+
+    var normal: [500*3]f32 = undefined;
+    var temp_vertices: [(500*3)+3]f32 = undefined;
+
+    std.mem.set(f32, &normal, 0.0);
+    std.mem.set(f32, &temp_vertices, 0.0);
+
+    temp_vertices[0] = 0.0;
+    temp_vertices[1] = (height/2.0);
+    temp_vertices[2] = 0.0;
+
+    temp_vertices[3] = 0.0;
+    temp_vertices[4] = -((height/2.0));
+    temp_vertices[5] = 0.0;
+
+    const float_num_tries = @intToFloat(f32, num_sides);
+    for (1..num_sides+1) |i|{
+        var float_i = @intToFloat(f32, i);
+        //6.283184 = (360*(3.14/180))
+        var x: f32 = radius * @sin((6.283184/float_num_tries)*float_i);
+        var y: f32 = radius * @cos((6.283184/float_num_tries)*float_i);
+
+        temp_vertices[i*6] = x;
```

Comment:
Why are you not storing this directly to vertex_data? Is there a special reason for this?
Because this is all the vertices that you need.

Reply:
See: https://github.com/hexops/mach-examples/pull/41#discussion_r1145060061